src/main.rs

```diff
@@ -29,6 +36,26 @@ macro_rules! upgrade_weak {
     };
 }
 
+#[derive(Deserialize, Serialize, Debug)]
+struct SnapshotSettings {
+    /// By default, the current one.
+    folder: PathBuf,
+    /// Either "BMP", "JPEG" or "PNG".
+    format: String,
```

Comment:
Why not an enum? And no BMP please, we have no writing support for that in GStreamer :)

Reply:
I'll need to check how serde works with enum but yes, good idea!

Reply:
With `serde-json` it just works :)